### ISSUE TITLE
NPE using payara-micro maven plugin 1.0.1 (bundle)

### ISSUE THREAD

**Issue by kengengler** [payara/ecosystem-maven#52]
[micro-profile-bundle-exc.txt](https://github.com/payara/ecosystem-maven/files/2280373/micro-profile-bundle-exc.txt)
I'm trying to use the maven micro plugin to build an uber jar. But I'm always ending up with a failed build due to an NPE. I've started with an application generated by the archetype:
  `mvn archetype:generate -DarchetypeGroupId=fish.payara.maven.archetypes -DarchetypeArtifactId=payara-micro-maven-archetype -DarchetypeVersion=1.0.1 -DgroupId=fish.payara.micro -DartifactId=micro-sample -Dversion=1.0-SNAPSHOT -Dpackage=fish.payara.micro.sample -Darchetype.interactive=false`
Then I perform a 
  `mvn install`
to place the WAR in my local maven repo.
And, next I try to build the uber jar with
 `mvn payara-micro:bundle`
Which works up to a point:

```
[INFO] Scanning for projects...
[INFO]
[INFO] -------------------< fish.payara.micro:micro-sample >-------------------
[INFO] Building micro-sample 1.0-SNAPSHOT
[INFO] --------------------------------[ war ]---------------------------------
[INFO]
[INFO] --- payara-micro-maven-plugin:1.0.1:bundle (default-cli) @ micro-sample ---
[INFO] Configured Artifact: fish.payara.extras:payara-micro:5.182:jar
[INFO] Unpacking /Users/gengler/Documents/.m2/repository/fish/payara/extras/payara-micro/5.182/payara-micro-5.182.jar to /Users/gengler/Work/git/micro-sample/target/extracted-payaramicro with includes "" and excludes ""
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/gengler/Work/git/micro-sample/src/main/resources
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/gengler/Work/git/micro-sample/src/main/resources
[INFO] Configured Artifact: fish.payara.micro:micro-sample:1.0-SNAPSHOT:war
[INFO] Copying micro-sample-1.0-SNAPSHOT.war to /Users/gengler/Work/git/micro-sample/target/extracted-payaramicro/MICRO-INF/deploy/ROOT.war
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1.494 s
[INFO] Finished at: 2018-08-10T16:34:30-07:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal fish.payara.maven.plugins:payara-micro-maven-plugin:1.0.1:bundle (default-cli) on project micro-sample: Execution default-cli of goal fish.payara.maven.plugins:payara-micro-maven-plugin:1.0.1:bundle failed.: NullPointerException -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException

```
If I run maven with debug and exception info logged, I see that the NPE at:
```
Caused by: java.lang.NullPointerException
    at fish.payara.maven.plugins.micro.processor.ArtifactDeployProcessor.handle (ArtifactDeployProcessor.java:103)
    at fish.payara.maven.plugins.micro.processor.BaseProcessor.gotoNext (BaseProcessor.java:77)

```
I think its trying to get the file from the artifact. 

Attached is the full maven output from the `-X -e` run

**Comment by OndrejM** [comment]
Hi, as I replied on the Payara forum, it should work if you run the plugin with a full build, e.g.:

```
mvn package payara-micro:bundle
```

The goal doesn't expect that you run it without building the WAR file first, and then fails because Maven doesn't provide some info that's needed.

I've created #51 to fix the exception, but even with this fix I'd recommend to run the bundle goal from within a full maven lifecycle and not separately.

**Comment by mulderbaba** [comment]
Fixed by #51
